### PR TITLE
Remove unnecessary warnings for LayoutKind.Auto and Sequential

### PR DESCRIPTION
**Pull request by ForNeVeR**
According to ECMA-335, only `LayoutKind.Explicit` may produce unverifiable code. So, this PR removes the unncessary warnings for `LayoutKind.Auto` and `Sequential`.

See fsharp/fslang-suggestions#596.

Not sure if I did the right thing with `fsharpqa`, but all the tests are fixed.

### CODE REVIEW

File: src/fsharp/TypeChecker.fs [hunk 1]
```diff
@@ -15030,7 +15030,9 @@ module EstablishTypeDefinitionCores =
             let hasMeasureableAttr = HasFSharpAttribute cenv.g cenv.g.attrib_MeasureableAttribute attrs
             let hasCLIMutable = HasFSharpAttribute cenv.g cenv.g.attrib_CLIMutableAttribute attrs
             
-            let hasStructLayoutAttr = HasFSharpAttribute cenv.g cenv.g.attrib_StructLayoutAttribute attrs
+            let structLayoutAttr = TryFindFSharpInt32Attribute cenv.g cenv.g.attrib_StructLayoutAttribute attrs
+            let hasStructLayoutAttr = Option.isSome structLayoutAttr
```
Comment (forki) [inline src/fsharp/TypeChecker.fs]: Can you please move these things down to usage. Maybe then you can get rid of isSome by using pattern matching

File: tests/fsharpqa/Source/Conformance/DeclarationElements/CustomAttributes/Basic/W_StructLayoutSequentialPos_ClassExpliCtr.fs [hunk 2]
```diff
@@ -1,15 +1,14 @@
-// #Regression #Conformance #DeclarationElements #Attributes 
+// #Regression #Conformance #DeclarationElements #Attributes
 // Regression: FSB 4014
 // Need to tighten up our imlementation of StructLayout.Sequential.
-//<Expects id="FS0009" span="(9,10-9,12)" status="warning">Uses of this construct may result in the generation of unverifiable \.NET IL code\. This warning can be disabled using '--nowarn:9' or '#nowarn "9"'\.</Expects>
 
-module PositiveTests = 
+module PositiveTests =
 
     [<System.Runtime.InteropServices.StructLayout(System.Runtime.InteropServices.LayoutKind.Sequential)>]
```
Comment (forki) [inline tests/fsharpqa/Source/Conformance/DeclarationElements/CustomAttributes/Basic/W_StructLayoutSequentialPos_ClassExpliCtr.fs]: Is there still a test for the remaining case?

Comment (ForNeVeR) [inline tests/fsharpqa/Source/Conformance/DeclarationElements/CustomAttributes/Basic/W_StructLayoutSequentialPos_ClassExpliCtr.fs]: Shouldn't we test that the code now compiles without any warning?

**Comment by forki** [comment]
I'm not sure if the system has this option. But I think it's fine. Just if
the remaining case is covered

Am 24.07.2017 7:59 vorm. schrieb "Friedrich von Never" <
notifications@github.com>:

File: tests/fsharpqa/Source/Conformance/DeclarationElements/CustomAttributes/Basic/W_StructLayoutSequentialPos_ClassExpliCtr.fs (hunk 2 above)
Comment (forki) [inline tests/fsharpqa/Source/Conformance/DeclarationElements/CustomAttributes/Basic/W_StructLayoutSequentialPos_ClassExpliCtr.fs]: In the corresponding env.lst file you can set compiler options. Please set --warningsaserrors (or how is it called?)

**Comment by ForNeVeR** [comment]
Yes, the remaining case with `LayoutKind.Explicit` is covered by `W_StructLayoutExplicit01.fs`.

I'm going to let other files in `fsharpqa` in their current state if that's okay.

File: tests/fsharpqa/Source/Conformance/DeclarationElements/CustomAttributes/Basic/W_StructLayoutSequentialPos_ClassExpliCtr.fs (hunk 2 above)
Comment (ForNeVeR) [inline tests/fsharpqa/Source/Conformance/DeclarationElements/CustomAttributes/Basic/W_StructLayoutSequentialPos_ClassExpliCtr.fs]: Done!

**Pull request merged**